modules/table/src/integrationTest/java/org/apache/ignite/distributed/ITTablePersistenceTest.java

```diff
@@ -125,14 +130,19 @@
 
     /** {@inheritDoc} */
     @Override public Path getListenerPersistencePath(PartitionListener listener) {
-        return ((RocksDbStorage) listener.getStorage()).getDbPath();
+        return paths.get(listener);
     }
 
     /** {@inheritDoc} */
     @Override public RaftGroupListener createListener(Path workDir) {
-        return new PartitionListener(
-            new RocksDbStorage(workDir.resolve(UUID.randomUUID().toString()), ByteBuffer::compareTo)
-        );
+        if (paths.containsValue(workDir))
```

Comment:
Looks like you search for the value twice, can we unify this check with the line below?

Reply:
For example:
```
return paths.entrySet()
    .stream()
    .filter(entry -> entry.getValue().equals(workDir))
    .findAny()
    .map(Map.Entry::getKey)
    .orElseGet(() -> {
        PartitionListener listener = new PartitionListener(new ConcurrentHashMapStorage());

        paths.put(listener, workDir);

        return listener;
    });
```